stl/inc/utility

```diff
@@ -712,21 +712,53 @@ _NODISCARD constexpr bool cmp_greater_equal(const _Ty1 _Left, const _Ty2 _Right)
 }
 
 // FUNCTION TEMPLATE in_range
-template <class _Ty1, class _Ty2>
-_NODISCARD constexpr bool in_range(const _Ty2 _Value) noexcept {
-    static_assert(_Is_standard_integer<_Ty1> && _Is_standard_integer<_Ty2>,
-        "The integer comparison functions only accept standard and extended integer types.");
-    using _UTy1 = make_unsigned_t<_Ty1>;
-    // _U_max is numeric_limits<_Ty1>::max() converted to _UTy1
-    constexpr auto _U_max = static_cast<_UTy1>(static_cast<_UTy1>(-1) >> static_cast<int>(is_signed_v<_Ty1>));
-    if constexpr (is_signed_v<_Ty1> == is_signed_v<_Ty2>) {
-        return static_cast<_Ty1>(_U_max + 1) <= _Value && _Value <= static_cast<_Ty1>(_U_max);
-    } else if constexpr (is_signed_v<_Ty2>) {
-        return _Value >= 0 && static_cast<make_unsigned_t<_Ty2>>(_Value) <= static_cast<_Ty1>(_U_max);
+template <class _Ty>
+_NODISCARD constexpr _Ty _Min_limit() noexcept { // same as (numeric_limits<_Ty>::min)(), less throughput cost
+    static_assert(_Is_standard_integer<_Ty>); // doesn't attempt to handle all types
+    if constexpr (is_signed_v<_Ty>) {
+        constexpr auto _Unsigned_max = static_cast<make_unsigned_t<_Ty>>(-1);
+        return static_cast<_Ty>((_Unsigned_max >> 1) + 1); // well-defined, N4861 [conv.integral]/3
     } else {
-        return _Value <= _U_max;
+        return 0;
     }
 }
+
+template <class _Ty>
+_NODISCARD constexpr _Ty _Max_limit() noexcept { // same as (numeric_limits<_Ty>::max)(), less throughput cost
+    static_assert(_Is_standard_integer<_Ty>); // doesn't attempt to handle all types
+    if constexpr (is_signed_v<_Ty>) {
+        constexpr auto _Unsigned_max = static_cast<make_unsigned_t<_Ty>>(-1);
+        return static_cast<_Ty>(_Unsigned_max >> 1);
+    } else {
+        return static_cast<_Ty>(-1);
+    }
+}
+
+template <class _Rx, class _Ty>
+_NODISCARD constexpr bool in_range(const _Ty _Value) noexcept {
+    static_assert(_Is_standard_integer<_Rx> && _Is_standard_integer<_Ty>,
+        "The integer comparison functions only accept standard and extended integer types.");
+
+    constexpr auto _Ty_min = _Min_limit<_Ty>();
+    constexpr auto _Rx_min = _Min_limit<_Rx>();
+
+    if constexpr (_STD cmp_less(_Ty_min, _Rx_min)) {
+        if (_Value < static_cast<_Ty>(_Rx_min)) {

```

Comment:
`_Rx_min` is a constant expression, and its conversion to `_Ty` is non-narrowing. We typically document and enforce that with `{}` conversion: (This is probably not worth resetting testing.)
```suggestion
        if (_Value < _Ty{_Rx_min}) {
```

Reply:
This is a great idea, and worth resetting testing! (Realizing that converting the limit would be non-narrowing is what led to this refactoring.)